src/Chapter2.hs

```diff
@@ -605,7 +630,11 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate =
+  foldr' go []
+  where
+    go x acc = x:x:acc
```

Comment:
Interesting implementation using standard functions! Though, using `foldr'` instead of plain `foldr` won't get many benefits here :slightly_smiling_face: